src/librustc_mir/build/matches/mod.rs

```diff
@@ -62,24 +61,28 @@ impl<'a,'tcx> Builder<'a,'tcx> {
         // highest priority candidate comes last in the list. This the
         // reverse of the order in which candidates are written in the
         // source.
-        let candidates: Vec<Candidate<'tcx>> =
-            arms.iter()
-                .enumerate()
-                .rev() // highest priority comes last
-                .flat_map(|(arm_index, arm)| {
-                    arm.patterns.iter()
-                                .rev()
-                                .map(move |pat| (arm_index, pat.clone(), arm.guard.clone()))
-                })
-                .map(|(arm_index, pattern, guard)| {
-                    Candidate {
+        let candidates: Vec<Candidate<'tcx>> = arms.iter()
+                                                   .enumerate()
+                                                   .rev()
+                                                   .flat_map(|(arm_index, arm)| {
+                                                       arm.patterns
+                                                          .iter()
+                                                          .rev()
+                                                          .map(move |pat| {
+                                                              (arm_index,
+                                                               pat.clone(),
```

Comment:
In general, this code is now taking up a LOT more space than it used to. I think we should revert this particular call-chain.